subprojects/core/src/test/groovy/org/gradle/api/internal/plugins/ImperativeOnlyPluginTargetTest.groovy

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.plugins
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.initialization.Settings
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.api.problems.internal.DefaultProblems
+import org.gradle.api.problems.internal.ProblemEmitter
+import org.gradle.internal.operations.CurrentBuildOperationRef
+import org.gradle.internal.operations.OperationIdentifier
+import spock.lang.Specification
+
+class ImperativeOnlyPluginTargetTest extends Specification {
+
+    def problemEmitter = Mock(ProblemEmitter)
+    def currentBuildOperationRef = Mock(CurrentBuildOperationRef) {
+        getId() >> new OperationIdentifier(42)
+    }
+    def problems = new DefaultProblems(problemEmitter, currentBuildOperationRef)
```

Comment:
🤔 This is a lot of setup currently. 
I think we should simplify mocking the service.

cc. @gradle/bt-ide-experience

Reply:
I think what's missing is a Problem API fixture for unit tests. That would be handy